python/mxnet/pretrained_model.py

```diff
@@ -0,0 +1,363 @@
+import os as _os
+import logging as _logging
+import json as _json
+from collections import namedtuple as _namedtuple
+from .model import FeedForward as _FeedForward
+from .model import extract_feature as _extract_feature
+from . import ndarray as _ndarray
+from . import io as _io
+import numpy as _np
+import requests
+
+_LOGGER = _logging.getLogger(__name__)
+
+class ModelTypeEnum(object):
+    """
+    Enumeration of Pretrained Model Type
+    """
+    IMAGE_CLASSIFIER= 'IMAGE_CLASSIFIER'
+
+ModelEntry = _namedtuple('ModelEntry', ['name', 'type', 'version'])
+
+
+"""
+Default location for model download
+"""
+DEFAULT_MODEL_LOCATION=_os.path.expanduser('~/.graphlab/mxnet_models')
+
+
+def list_models(location=DEFAULT_MODEL_LOCATION):
+    """
+    Return list of pretrained model names.
+
+    Parameters
+    ----------
+    location : str, optional
+      The local directory where the model is saved to.
+
+    Examples
+    --------
+    >>> mx.pretrained_model.list_models()
+    """
+    if not _os.path.exists(location):
+        _os.makedirs(location)
+
+    models = [p for p in _os.listdir(location)]
+    ret = []
+    for name in models:
+        model_path = _os.path.join(location, name)
+        if not _os.path.isdir(model_path):
+            continue
+        metadata_path = _os.path.join(model_path, 'metadata.json')
+        version = None
+        try:
+            f = open(metadata_path)
+            metadata = _json.load(f)
+            version = metadata['version']
+            model_type = metadata['model_type']
+            name = metadata['name']
+            ret.append(ModelEntry(name, model_type, version))
+        except Exception as e:
+            _LOGGER.warn('Unable to open or parse model metadata %s.' % metadata_path)
+    return ret
+
+def _download_file(url, target_file):
+    r = requests.get(url, stream=True)
+    with open(target_file, 'wb') as f:
+        for chunk in r.iter_content(chunk_size=1024 * 1024):
+            if chunk:
+                f.write(chunk)
+
+def download_model(url, location=DEFAULT_MODEL_LOCATION, overwrite=False):
+    """
+    Perform downloading the model to local filesystem.
+
+    Parameters
+    ----------
+    url : str,
+      URL of the model. Get the list of available models from https://github.com/dato-code/mxnet.
```

Comment:
It would also be cool to be able to get available models programmatically, via API call in interactive python session.